tools/ci_connector_ops/ci_connector_ops/pipelines/pipelines/metadata.py

```diff
@@ -212,3 +262,34 @@ async def run_metadata_upload_pipeline(
             pipeline_context.test_report = TestReport(pipeline_context, results)
 
         return pipeline_context.test_report.success
+
+
+async def run_metadata_orchestrator_deploy_pipeline(
+    is_local: bool,
+    git_branch: str,
+    git_revision: str,
+    gha_workflow_run_url: Optional[str],
+    pipeline_start_timestamp: Optional[int],
+    ci_context: Optional[str],
+) -> bool:
+    metadata_pipeline_context = PipelineContext(
+        pipeline_name="Metadata Service Orchestrator Unit Test Pipeline",
+        is_local=is_local,
+        git_branch=git_branch,
+        git_revision=git_revision,
+        gha_workflow_run_url=gha_workflow_run_url,
+        pipeline_start_timestamp=pipeline_start_timestamp,
+        ci_context=ci_context,
+    )
+
+    async with dagger.Connection(DAGGER_CONFIG) as dagger_client:
+        metadata_pipeline_context.dagger_client = dagger_client.pipeline(metadata_pipeline_context.pipeline_name)
+
+        async with metadata_pipeline_context:
+            steps = [
+                TestOrchestrator(context=metadata_pipeline_context),
+                DeployOrchestrator(context=metadata_pipeline_context)
+            ]
+            steps_results = await run_steps(steps)
```

Comment:
@alafanechere I addressed your comments and this is back up and ready for review.

Though there is one net new function added that you might recognize from [here](https://github.com/airbytehq/airbyte/pull/24225#discussion_r1162064279).

I added it as the logic was useful for "dont run next steps" and should be able to be moved into the pipeline class once the pipeline also contains the steps.